qa/workunits/rbd/compare_mirror_images.sh

```diff
@@ -0,0 +1,154 @@
+#!/bin/bash
+
+set -ex
+
+IMG_PREFIX=image-
+RBD_MIRROR_MODE=snapshot
+MNTPT_PREFIX=/mnt/test
+WORKLOAD_TIMEOUT=5m
+
+. $(dirname $0)/rbd_mirror_helpers.sh
+
+launch_manual_msnaps() {
+  local cluster=$1
+  local pool=$2
+  local image=$3
+
+  for i in {1..30}; do
+    mirror_image_snapshot $cluster $pool $image
+    sleep 3s;
+  done
+}
+
+run_bench() {
+  local mountpt=$1
+  local timeout=$2
+
+  KERNEL_TAR_URL="https://cdn.kernel.org/pub/linux/kernel/v4.x/linux-4.14.280.tar.gz"
+  sudo wget $KERNEL_TAR_URL -O $mountpt/kernel.tar.gz
+  sudo timeout $timeout bash -c "tar xvfz $mountpt/kernel.tar.gz -C $mountpt \
+    | pv -L 1k --timer &> /dev/null" || true
+}
+
+wait_for_demote_snap () {
+  local cluster=$1
+  local pool=$2
+  local image=$3
+
+  for s in 1 2 4 8 8 8 8 8 8 8 8 16 16; do
+    RET=$(rbd --cluster $cluster snap ls --all $pool/$image | tail -n 1 \
+            | grep non_primary | grep demote | grep -v "%" || true)
+    if [ "$RET" != "" ]; then
+      echo demoted snapshot received, continuing
+      sleep 30s # wait a bit for it to propagate
```

Comment:
sleeptime was 10s in the original PR https://github.com/ceph/ceph/pull/52132 . Increased it to 30s to check whether it helps with MD5 hashes failing to match for only nbd workloads. Still fails,
 https://pulpito.ceph.com/rraja-2023-12-06_01:23:15-rbd:mirror-main-distro-default-smithi/

Reply:
Got it!